Sn 100 add responsiveness to slider

MariamBashir opened the pull request:
Add responsiveness to the slider in Axiom. This was creating problems for the Bubble chart component as the slider did adjust its width with the window resizing. 

Before

<img width="580" alt="Screen Shot 2019-08-15 at 14 39 53" src="https://user-images.githubusercontent.com/38217242/63095070-90e6e380-bf6a-11e9-8ce1-13455d071506.png">

After
<img width="606" alt="Screen Shot 2019-08-15 at 14 40 11" src="https://user-images.githubusercontent.com/38217242/63095100-a2c88680-bf6a-11e9-93ac-02e8f2f88c87.png">

clairebw commented on packages/axiom-components/src/Slider/Range.js:
```diff
@@ -197,55 +219,82 @@ export default class Range extends Component {
+    }
+
+    const { width: trackWidth } = track.getBoundingClientRect();
+    this.setState({ trackWidth });
```
When we move a component to have the dashboard split/unsplit the Customize Bar is rerendered and so is the slider. We end up having a new `trackWidth` after the render with no window 'resize' event being triggered. 

Is there a way to handle this? or is this case already addressed? I'm not sure.

lpoulter replied:
Not sure I fully understand @clairebw 's use case but it sounds like you should be able to use [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). This will trigger when the element changes size, it's also more efficient as it does not trigger on every window resize. You will need to use the polyfill for edge support.

lilybrandwatch commented:
@MariamBashir It's hard to judge with the screenshots only but in your "After", the inputs on the left needs to be resized too which will allowed more space to the slider like it in my design but I guess this is because you were only focusing on the sliders . For the rest if the handle size works instead of the input I'm fine with this as solution but I'd like to test it first of course. 
Thanks. :)

lpoulter commented on packages/axiom-components/src/Slider/Range.js:
```diff
@@ -133,6 +150,11 @@ export default class Range extends Component {
 
+  componentDidMount(){
+    this.handleResize();
+    window.addEventListener('resize', this.handleResize);
```
Not sure this Listener is removed, but you might want to use ResizeObserver anyway, see below.

lpoulter commented on packages/axiom-components/src/Slider/Range.js:
```diff
@@ -197,55 +219,82 @@ export default class Range extends Component {
+              onFocus={ disabled ? null : this.handleFocus }
+              onMouseLeave={ () => this.setState({ isMouseOver: false }) }
+              onMouseOver={ () => this.setState({ isMouseOver: true }) }
+              style={ { maxWidth: MAX_SLIDER_WIDTH } }
```
This max width seems to be causing issues when it's parent, in this case a `GridCell`, is larger than this max width. A gap is created between the label and the track.

lpoulter replied:
<img width="287" alt="Screen Shot 2019-08-28 at 21 05 25" src="https://user-images.githubusercontent.com/3940567/63888809-99fa9a80-c9d7-11e9-92b0-4e329062db4c.png">

MariamBashir replied:
Interesting, ill dig into that.

MariamBashir commented:
@lpoulter I will do some research on the `ResizeObserver` and replace the window resize event with that. Thank you for the feedback.

@lilybrandwatch I will implement the changes and then set up a meeting with you again so you can test it with me and we can take it from there. Thanks for taking a look.

MariamBashir commented:
@lpoulter , I have added the resizeObserver which also covers the case @clairebw mentioned. Also removed the `MAX_WIDTH` which was causing the min label to appear away from the slider track. 

@lilybrandwatch as discussed yesterday with you, i will create a new ticket for resizing the drop downs in the customize bar.  This ticket focused on the sliders and makes its responsive based on the width of the customize bar. I have created another [ticket](https://office.brandwatch.com/jira/browse/SN-192) to add responsiveness to the drop downs based on the width of customize bar.
